### PR TITLE
Init and shutdown logging

### PR DESCRIPTION
**Pull request by bhalevy**
Increase visibility of startup and shutdown messages.

Ref #4384

### CODE REVIEW

**Comment by espindola** [comment]
I am not sure when to use startlog.info or
supervisor::notify. supervisor::notify Also sends the message to
systemd, is that it? But the logs are also in systemd anyway.

Other than that, LGTM

**Comment by avikivity** [comment]
@espindola `supervisor::notify` messages are visible in `systemctl status scylla-server`

File: main.cc [hunk 1]
```diff
@@ -461,6 +461,7 @@ int main(int ac, char** av) {
                 pctx.prefix = cfg->prometheus_prefix();
                 prometheus_server.start("prometheus").get();
                 stop_prometheus = ::make_shared(defer([&prometheus_server] {
+                    startlog.info("stopping prometheus API server");
```
[the remark this answers is not in the document]
Comment (nyh) [reply main.cc]: You can have a generic stop(name, func) function that prints "stopping name", runs func() and then a finally() to print "stopped name", to make this all automatic with minimum boilerplate code.

[the remark this answers is not in the document]
Comment (bhalevy) [reply main.cc]: @nyh good idea, let's open a follow up issue to do that enhancement.

**Pull request merged**